tools/driver/p4c/main.py

```diff
@@ -0,0 +1,190 @@
+#!/usr/bin/env python
+
+"""
+p4c - P4 Compiler Driver
+
+"""
+
+from __future__ import absolute_import
+import argparse
+import glob
+import itertools
+import os
+import subprocess
+import sys
+
+import p4c.run
+import p4c.util
+import p4c.config
+
+commands = {}
+
+def init_globals():
+    commands['preprocessor'] = []
+    commands['compiler'] = []
+    commands['assembler'] = []
+    commands['linker'] = []
+
+def save_options(cmd, options):
+    for opt in options:
+        opt_split = opt.split(' ')
+        for s in opt_split:
+            commands[cmd].append(s)
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-v", "--version", dest="show_version",
+                        help="show version and exit",
+                        action="store_true", default=False)
+    parser.add_argument("-debug", dest="debug",
+                        help="verbose",
+                        action="store_true", default=False)
+    parser.add_argument("-###", dest="dry_run",
+                        help="print (but do not run) the commands",
+                        action="store_true", default=False)
+    parser.add_argument("-Xpreprocessor", dest="preprocessor_options",
+                        metavar="<arg>",
+                        help="Pass <arg> to the preprocessor",
+                        action="append", default=[])
+    parser.add_argument("-Xp4c", dest="compiler_options",
+                        metavar="<arg>",
+                        help="Pass <arg> to the compiler",
+                        action="append", default=[])
+    parser.add_argument("-Xassembler", dest="assembler_options",
+                        metavar="<arg>",
+                        help="Pass <arg> to the assembler",
+                        action="append", default=[])
+    parser.add_argument("-Xlinker", dest="linker_options",
+                        metavar="<arg>",
+                        help="Pass <arg> to the linker",
+                        action="append", default=[])
+    parser.add_argument("-b", dest="backend",
```

Comment:
It's also worth noting that the include path for different architectures may be different, so we may want to let the configurations determine it.